src/code_translation/ir/nodes.py

```diff
@@ -176,82 +217,61 @@ def sql_string(self) -> Optional[str]:
             join_operator = f"{self.how.upper()} {join_operator}"
 
         # Extract query and additional information from left node
-        left_set_key = False
-        if isinstance(self.left, SQLNode):
-            left_sql = self.left.sql_string
-        elif isinstance(self.left, JoinNode):
-            left_sql = self.left.sql_string
-        elif isinstance(self.left, AggregationNode):
-            # This is where the dependency injection starts to get complicated
-            # TODO: find out if there is a precode condition for the CSTTranslation
-            left_cst_translation = self.left.to_cst_translation(sql_access_method="")  #
-            if left_cst_translation.precode != []:
-                raise Exception("No support for prequeries in JoinNodes yet")  # TODO: Investigate this further...
-            left_sql = self.left.sql_string
-        elif isinstance(self.left, SetKeyNode):
-            left_set_key = True
-            left_sql = self.left.node.sql_string
-            left_key = self.left.key
+
+        left_sql = self.left.sql_string
+        left_on: list[str]
+        if self.left_on == "key":
+            assert isinstance(self.left, SetKeyNode)  # TODO: Don't do that
```

Comment:
the way to avoid this is by adding a property "key" to every DataFrameNode which for SetKeyNodes is the key and for most other nodes is just the key of the child